### PR TITLE
[function](hash) add support of murmur_hash3_64

### PR DESCRIPTION
**Pull request by jacktengg**
Also delete duplicated code for murmur_hash3_32

# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by winnerhero — COMMENTED** [review]
LGTM

File: be/src/util/hash_util.hpp [hunk 1]
```diff
@@ -132,60 +132,11 @@ class HashUtil {
...
-        h ^= h >> 16;
-        h *= 0x85ebca6b;
-        h ^= h >> 13;
-        h *= 0xc2b2ae35;
-        h ^= h >> 16;
-        return h;
-    }
-
     // modify from https://github.com/aappleby/smhasher/blob/master/src/MurmurHash3.cpp
     static uint32_t murmur_hash3_32(const void* key, int32_t len, uint32_t seed) {
```
Comment (yiguolei) [inline be/src/util/hash_util.hpp:136]: Check the comment: modify from https://github.com/aappleby/smhasher/blob/master/src/MurmurHash3.cpp

Why modify the origin code in the past? Is the modified method the same as murmur_hash3_x86_32?

Comment (jacktengg) [reply be/src/util/hash_util.hpp:136]: The modified method is the same as murmur_hash3_x86_32, just different in returning value.

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**